plugin/pkg/auth/authorizer/node/node_authorizer.go

```diff
@@ -309,13 +309,21 @@ func (r *NodeAuthorizer) authorizeResourceSlice(nodeName string, attrs authorize
 		return authorizer.DecisionNoOpinion, "cannot authorize ResourceSlice subresources", nil
 	}
 
-	// allowed verbs: get, create, update, patch, delete
+	// allowed verbs: get, create, update, patch, delete, watch, list, deletecollection
 	verb := attrs.GetVerb()
 	switch verb {
 	case "get", "create", "update", "patch", "delete":
 		// Okay, but check individual object permission below.
```

Comment:
move the individual checks into the switch cases... doing nothing here and having the fallthrough below is confusing:

```go
case "create":
	// For create, the noderestriction admission plugin is performing this check.
	// Here we don't have access to the content of the new object.
  return authorizer.DecisionAllow, "", nil

case "get", "update", "patch", "delete":
	// The request must come from a node with the same name as the ResourceSlice.NodeName field.
	// The existing object must have established that access is allowed by recording a graph edge.
	return r.authorize(nodeName, sliceVertexType, attrs)

case "watch", "list", "deletecollection":
  ...
```

Reply:
Done.